android/src/main/java/com/bitmovin/player/reactnative/RNPlayerView.kt

```diff
@@ -94,11 +95,30 @@ private val EVENT_CLASS_TO_REACT_NATIVE_NAME_MAPPING_UI = mapOf<KClass<out Event
  * exposes player events as bubbling events.
  */
 @SuppressLint("ViewConstructor")
-class RNPlayerView(val context: ReactApplicationContext) :
-    LinearLayout(context),
-    LifecycleEventListener,
-    View.OnLayoutChangeListener,
-    RNPictureInPictureDelegate {
+class RNPlayerView(
+    private val context: ReactApplicationContext,
+) : LinearLayout(context), View.OnLayoutChangeListener, RNPictureInPictureDelegate {
+    private val activityLifecycle = (context.currentActivity as ReactActivity).lifecycle
```

Comment:
Is this safe to call? `currentActivity` is nullable and of type `Activity`.

Reply:
In my opinion yes.
- `currentActivity` is nullable because it's a function in application context scope and it's a legit use case to have no activity in some scenarios in that scope. But we are attaching a view here, so if there is no activity attached and we are creating a PlayerView we are probably doing something wrong.
- Regarding activity type a ReactActivity is needed to create RN root view, so this should also be safe. The actual cast that we need here is `AppCompatActivity`, but since RectActivity extends `AppCompatActivity` I thought this was somewhat cleaner.

What we could do is to handle it in a more descriptive way or even go very safe and instead of throwing we could just log errors - in that case something will likely be broken though. I think I prefer throwing, since I don't see a legit use case where the casting wouldn't work but maybe I am missing something of course.

Can you think of a use case where this casting wouldn't work?

Reply:
We now throw an IllegalStateException with a clear message if the casting fails: https://github.com/bitmovin/bitmovin-player-react-native/pull/313/commits/bdcb1203bf00eb55e1e775ad8ef1402349768cdf